src/execute/index.js

```diff
@@ -327,7 +327,9 @@ function oas3BaseUrl({ spec, pathName, method, server, contextUrl, serverVariabl
 }
 
 function buildOas3UrlWithContext(ourUrl = '', contextUrl = '') {
-  const parsedUrl = url.parse(ourUrl);
+  // Relative server url should be resolved against contextUrl
+  const parsedUrl =
```

Comment:
Wouldn't this simplified code produce the same result? We always resolve before we parse. 

```js
const parsedUrl = url.parse(url.resolve(contextUrl, ourUrl));
```

Reply:
With the simplied code, `serverUrl=''` and `contextUrl='http://t.com/doc/api.json'` will produce `'http://t.com/doc/api.json'` while it previously produces `'http://t.com'`. This might not be expected.

Reply:
Right.